src/lambda/commands/deleteLambda.ts

```diff
@@ -5,37 +5,63 @@
 
 'use strict'
 
-import * as nls from 'vscode-nls'
-const localize = nls.loadMessageBundle()
+import { OutputChannel, window } from 'vscode'
+import { loadMessageBundle } from 'vscode-nls'
+const localize = loadMessageBundle()
 
 import { LambdaClient } from '../../shared/clients/lambdaClient'
-import { ext } from '../../shared/extensionGlobals'
 import { StandaloneFunctionNode } from '../explorer/standaloneNodes'
 
-export async function deleteLambda(
-    node: StandaloneFunctionNode,
-    refresh: () => void
-): Promise<void> {
-    const client: LambdaClient = ext.toolkitClientBuilder.createLambdaClient(node.regionCode)
+export async function deleteLambda({
+                           lambdaClient,
+                           node,
+                           outputChannel,
+                           onRefresh,
+                           onConfirm = async () => {
```

Comment:
I don't think that this is the right level of mockability--we only want to mock `window.showWarningMessage`, the rest of `onConfirm` still needs test coverage.

Reply:
Our need to mock can be greatly reduced by writing less brittle, tightly coupled code. Thinking in a non-tightly coupled way, we are inserting logic to support confirmation. It might be sending and SMS, receiving and verifying the associated security token, it may be a capcha... the implementation really doesn't matter. What does matter is that we are adding support for a function that enables confirm or deny by simply returning true of false.

See "Mocking as a code smell": https://medium.com/javascript-scene/mocking-is-a-code-smell-944a70c90a6a

Yes we could test `vscode.window.showWarningMessage` as part of our integ tests, but the need to mock it only stems from tightly coupled code.